packages/ui/src/hooks/utils.ts

```diff
@@ -0,0 +1,110 @@
+import { ConnectorData } from '@/types';
+import { generateRandomString } from '@/utils';
+
+/**
+ * Native SDK Utility Methods
+ */
+export const getLogtoNativeSdk = () => {
+  if (typeof logtoNativeSdk !== 'undefined') {
+    return logtoNativeSdk;
+  }
+};
+
+export const isNativeWebview = () => {
+  const platform = getLogtoNativeSdk()?.platform ?? '';
+
+  return ['ios', 'android'].includes(platform);
+};
+
+/**
+ * Social Connector State Utility Methods
+ * @param state
+ * @param state.uuid - unique id
+ * @param state.platform - platform
+ * @param state.callbackLink - callback uri scheme
+ */
+
+type State = {
+  uuid: string;
+  platform: 'web' | 'ios' | 'android';
+  callbackLink?: string;
+};
+
+const storageKeyPrefix = 'social_auth_state';
+
+export const generateState = () => {
+  const uuid = generateRandomString();
+  const platform = getLogtoNativeSdk()?.platform ?? 'web';
+  const callbackLink = getLogtoNativeSdk()?.callbackLink;
+
+  const state: State = { uuid, platform, callbackLink };
+
+  return btoa(JSON.stringify(state));
+};
+
+export const decodeState = (state: string) => {
+  try {
+    return JSON.parse(atob(state)) as State;
+  } catch {}
+};
+
+export const stateValidation = (state: string, connectorId: string) => {
+  const stateStorage = sessionStorage.getItem(`${storageKeyPrefix}:${connectorId}`);
+
+  return stateStorage === state;
+};
+
+export const storeState = (state: string, connectorId: string) => {
+  sessionStorage.setItem(`${storageKeyPrefix}:${connectorId}`, state);
+};
+
+/**
+ * Social Connectors Filter Utility Methods
+ */
+export const filterSocialConnectors = (socialConnectors?: ConnectorData[]) => {
+  if (!socialConnectors) {
+    return [];
+  }
+
+  const connectorMap = new Map<string, ConnectorData>();
+
+  if (!isNativeWebview()) {
+    for (const connector of socialConnectors) {
+      const { platform, target } = connector;
+
+      if (platform === 'Native') {
+        continue;
+      }
+
+      if (platform === 'Web' || !connectorMap.get(target)) {
```

Comment:
nit: better add some comments, it takes me a bit time to understand.

Reply:
good point